src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1707,7 +1796,7 @@ void Lowering::ContainCheckMul(GenTreeOp* node)
 //
 void Lowering::ContainCheckDivOrMod(GenTreeOp* node)
 {
-    assert(node->OperIs(GT_DIV, GT_UDIV));
+    assert(node->OperIs(GT_DIV, GT_UDIV, GT_MOD));
```

Comment:
why not also `GT_UMOD`?

Reply:
This optimization doesn't impact GT_UMOD, only GT_MOD so that's why the additional GT_MOD check is there.